### DISCUSSION TITLE
`.connect()` vs. `.on()`

### DISCUSSION THREAD

**Discussion by musjj** [Aylur/ags#300] [Q&A] [answered]
I'm still not quite sure what's the difference between `.connect()` and `.on()`. The docs mentions:

> `on` is the same as `connect` but instead of the signal handler id, it returns a reference to the widget. `on` will setup a callback on a widget signal.

But both of them seems to be returning `self` (or something else?):

```typescript
const myVar = Variable(0);
myVar.connect("changed", (self) => print(self.value));

const myButton = Widget.Button();
myButton.connect("clicked", (self) => print(self.action_name));

const myButton = Widget.Button();
myButton.on("clicked", (self) => print(self.action_name));
```

**Comment by Aylur** [comment]
```js
const widget = Widget.Button()
const onReturn = widget.on("clicked", () => print("clicked"))
const connectReturn = widget.connect("clicked", () => print("clicked"))

onReturn === widget // true
typeof connectReturn // number

widget.disconnect(connectReturn) // used to disconnect
```

**Reply by musjj** [reply]
Thanks! So if I get this right, the difference is only in the return value? So both of these callbacks will be called with `self` right?

**Reply by Aylur** [reply, marked as answer]
yes, `.on()` just calls `.connect` with the same parameters actually

**Reply by musjj** [reply]
Thanks for clearing it up!
